### PR TITLE
Update component composition

### PR DESCRIPTION
**Pull request by patsy-issa**
The following pr updates the component-composition example:

- Ember infrastructure and idioms
- Adds a smoke test
- Re-enables ci tests for the directory

### CODE REVIEW

**Comment by patsy-issa** [comment]
Express is not running in tests in the CI/CLI checking it out

**Comment by patsy-issa** [comment]
So this appears to be a design decision, http-mocks will not run in test environments and the recommended path is to use pretender/mirage, @marcoow thoughts about switching out express with mirage? 

https://github.com/ember-cli/ember-cli/pull/1423

**Comment by marcoow** [comment]
The express server is for development only. For the tests we should use pretender as we do for the other exmples applications (not Mirage please as that would severely complicate things) – see e.g. https://github.com/simplabs/ember-workshop/blob/master/examples/core-concepts/tests/acceptance/songs-test.js#L10 for reference

File: examples/component-composition/app/components/list-view.js [hunk 1]
```diff
@@ -1,23 +1,26 @@
-import Component from '@ember/component';
```
Comment (marcoow) [inline examples/component-composition/app/components/list-view.js:1]: Let's default to having components in their own directories (`app/components/list-view/component.js`) – that's what I tell people to do in the slides as well now.

**Review by marcoow — COMMENTED** [review]
Looks good – just one thing: I think we should default to having components in folders in `app/components`, e.g. `app/components/list-view` – that's what I'm updating the slides to as well now and I think is the default in ember-cli now?

File: examples/component-composition/app/components/list-view.js (hunk 1 above)
Comment (patsy-issa) [reply examples/component-composition/app/components/list-view.js:1]: Moved components into their own directories

File: examples/component-composition/app/templates/songs.hbs [hunk 2]
```diff
@@ -1,8 +1,8 @@
-<h1>{{songCount}} songs</h1>
+<h1 class="song-count">{{this.songCount}} songs</h1>
 
-<ListView @items={{model}} @onRemoveItem={{action "deleteSong"}} as |list item|>
-  <SongTile @song={{item}} @onSelectSong={{list.selectItem}} />
-  {{#list.delete-button}}
+<ListView @items={{this.model}} @onRemoveItem={{fn this.deleteSong}} as |list item|>
```
Comment (marcoow) [inline examples/component-composition/app/templates/songs.hbs:3]: Is the `fn` needed here?

Comment (patsy-issa) [reply examples/component-composition/app/templates/songs.hbs:3]: Nice catch 👍 not needed and allows us to remove `@action` as well

**Review by marcoow — APPROVED** [review]

**Pull request merged**